src/main/java/com/exactpro/th2/FixHandler.java

```diff
@@ -560,7 +560,7 @@ private void recovery(int beginSeqNo, int endSeqNo) {
 
                     if(ADMIN_MESSAGES.contains(msgType)) return true;
                     FixField possDup = findField(buf, POSS_DUP_TAG);
-                    if(possDup != null && possDup.getValue() == IS_POSS_DUP) return true;
+                    if(possDup != null && Objects.equals(possDup.getValue(), IS_POSS_DUP)) return true;
```

Comment:
```suggestion
                    if(Objects.equals(possDup.getValue(), IS_POSS_DUP)) return true;
```

Reply:
possDup can be null, which will lead to null pointer during Object.equals call